machinery-helper/rpm/rpm.go

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) 2015 SUSE LLC
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of version 3 of the GNU General Public License as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, contact SUSE LLC.
+//
+// To contact SUSE about this file by physical or electronic mail,
+// you may find current contact information at www.suse.com
+
+package rpm
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"unicode/utf8"
+)
+
+// Parses an output line from the rpm command.
+func ParseRpmLine(line string) (fileType string, fileName string, linkTarget string) {
+	fileType = line[0:1]
+
+	index := strings.Index(line, "/")
+	if index < 0 {
+		panic(line)
+	}
+	file := line[index:]
+	fields := strings.Split(file, " -> ")
+	if len(fields) == 2 {
+		fileName = fields[0]
+		linkTarget = fields[1]
+	} else {
+		fileName = file
+	}
+	return
+}
+
+// Returns a list of rpms handled by the system.
+func GetRpms() []string {
+	cmd := exec.Command("rpm", "-qlav")
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err := cmd.Run()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	f := func(c rune) bool {
+		return c == '\n'
+	}
+	packages := strings.FieldsFunc(out.String(), f)
+	return packages
+}
+
+// Appends implicit managed dirs to a map.
+func AddImplicitlyManagedDirs(dirs map[string]bool, files map[string]string) {
+	for file, target := range files {
+		for i := 1; i < len(file); i++ {
+			if file[i] == '/' {
+				topdir := file[:i]
+				if _, ok := dirs[topdir]; !ok {
+					dirs[topdir] = false
+				}
+			}
+		}
+
+		if target != "" {
+			if _, ok := dirs[target]; ok {
+				dirs[file] = false
+			}
+		}
+	}
+	return
+}
+
+// Builds a json from a map of files.
```

Comment:
comment on exported function AssembleJSON should be of the form "AssembleJSON ..."